pointcloud2_to_gridmap:
  ros__parameters:

    # #For LIO-SAM
    # pointcloud_topic: "/lio_sam/mapping/cloud_registered"
    # map_frame_id: "odom"
    # base_frame_id: "base_link"
    # integration_time: 0.6

    # For FAST_LIO
    pointcloud_topic: "cloud_registered"
    map_frame_id: "odom"
    base_frame_id: "base_link"
    integration_time: 0.6


    filters:
      filter1:  # Fill holes in the map with inpainting.
        name: inpaint
        type: gridMapCv/InpaintFilter
        params:
          input_layer: elevation
          output_layer: elevation_inpainted
          radius: 0.1

      filter2:  # Compute surface normals.
        name: surface_normals
        type: gridMapFilters/NormalVectorsFilter
        params:
          input_layer: elevation_inpainted
          output_layers_prefix: normal_vectors_
          radius: 0.2
          normal_vector_positive_axis: z

      filter3:  # Compute slope from surface normal.
        name: slope
        type: gridMapFilters/MathExpressionFilter
        params:
          output_layer: slope
          expression: acos(normal_vectors_z)
